### DISCUSSION TITLE
"Not complete" warning in GnssSimulateReceiver

### DISCUSSION THREAD

**Discussion by laplace76** [groops-devs/groops#182] [Q&A] [answered]
I am getting this error when trying to simulate the Galileo observation for a static receiver, using GnssSimulateReceiver:

![image](https://user-images.githubusercontent.com/62875774/233399090-c83b89ff-b3b4-44d6-951b-1ba383f5c404.png)

Any idea about the warning "not complete" (I am using the same wildcard included in the GnssType description). Moreover I would like to know if it is necessary to specify the parametrization (I assume a default parametrization is defined) and if the preprocessing section in the receiver is silently ignored (I assume it is, I see no reason to preprocess simulated data).

With Kind Regards

**Comment by tmayerguerr** [comment]
Hi,

In the simulation case, you must specify all observation types in detail. Wildcards are not meaningful here. E.g. C1CG, C1WG, L1CG, L2WG for GPS.

Preprocessing considers only elevationCutOff and elevationTrackMinimum.

Parametrization can be set up to simulate [troposphere](https://groops-devs.github.io/groops/html/gnssParametrizationType.html#troposphere) delays and apriori [signalBiases](https://groops-devs.github.io/groops/html/gnssParametrizationType.html#signalBiases) and in future apriori ionospheric delays (not implemented at the moment).

regards
Torsten

**Comment by laplace76** [comment]
Dear Torsten,
Thank you for the reply, unfortunately I still get errors. For example, using only C1BE and L1BE I get this:
![image](https://user-images.githubusercontent.com/62875774/233590673-0f86e085-899d-4b27-b64f-d377e8b72efb.png)

As you can see from the screenshot, I am creating a fake antenna file for the ABMF station (antenna model TRM57971.00) because the (receiver) antennaDefinition_igs14.dat does not contain entries for galileo (in the fake file, the PCV/PCO values are set to 0).

The receiver object is instructed to throw an exception if the observation type is not found: this does not happen, so I guess the observation type is recognized (I am using C1BE and L1BE). 

As I am interested in simulating Galileo, does the simulation program support it or not?

**Comment by tmayerguerr** [comment, marked as answer]
Hi,

Ok, GnssSimulateReceiver is not very well documented and not easy to use.

GROOPS only works with receivers with at least two frequencies to determine the STEC. It discards all observations without at least two code and two phase observations at different frequencies each.

Torsten

**Reply by laplace76** [reply]
Dear Torsten,
thank you very much for the help, now it works.
